Libraries/Microsoft.Teams.Apps/App.cs

```diff
@@ -62,29 +62,18 @@ public App(AppOptions? options = null)
 
         Client = options?.Client ?? options?.ClientFactory?.CreateClient() ?? new Common.Http.HttpClient();
         Client.Options.AddUserAgent(UserAgent);
-        Client.Options.TokenFactory = () =>
+        Client.Options.TokenFactory ??= () =>
         {
-            if (Credentials is not null)
+            if (Credentials is not null && (Token is null || Token.IsExpired))
```

Comment:
PR to fix startup issues has been merged, u can remove this and merge from main instead.